java/lance-namespace-core/src/test/java/com/lancedb/lance/namespace/TestObjectIdentifier.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.lancedb.lance.namespace;
+
+import com.google.common.collect.Lists;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TestObjectIdentifier {
+  @Test
+  public void testObjectIdentifier() {
+    // Case 1: empty namespace
+    ObjectIdentifier oid = ObjectIdentifier.of(Lists.newArrayList());
+    assertEquals(0, oid.size());
+
+    List<String> ns = Lists.newArrayList();
+    ns.add(null);
+    ns.add("");
+    assertEquals(0, oid.size());
```

Comment:
I think you have not constructed an oid yet?

Reply:
Also for null and empty string, I think we should just fail the operation and say these are invalid inputs, rather than allowing it and remove it from the levels.